src/classes/PS_CommitmentRequest.cls

```diff
@@ -165,7 +193,15 @@ public inherited sharing class PS_CommitmentRequest {
             .withProductMetaData(productMetaData);
 
         setDonorInfo(reqBody, rd);
-        setSchedules(reqBody, rd.RecurringDonationSchedules__r, currencyCode);
+        setSchedules(reqBody, scheduleService.buildNewSchedules(rd), currencyCode);
+
+        return reqBody;
+    }
+
+    private RequestBody getUpdateRequestBody(npe03__Recurring_Donation__c rd, String token) {
```

Comment:
No need to build schedules when credit card is updated for an existing Elevate RD (W-8585272).